### ISSUE TITLE
Need ability to modify the li props in a MenuItem in order to create a proper a11y menu

### ISSUE THREAD

**Issue by bvandrc** [palantir/blueprint#5320]
#### Environment

- __Package version(s)__: 4.3.2
- __Browser and OS versions__: All

#### Feature request

Need a way to pass props to `MenuItem` that get assigned to the `li` element rather than being applied to the inner `a` element. Per the a11y menu example here: https://w3c.github.io/aria-practices/examples/menubar/menubar-navigation.html , I need to be able to set `<li role="none">`. The reasons are also given in this page:
- `<li role="none">` Removes the implied listitem role of the li element.
- Necessary because the parent ul is serving as a menu so the li elements are not in their required list context.

#### Examples

See "HTML Source Code" section of this page: https://w3c.github.io/aria-practices/examples/menubar/menubar-navigation.html

**Comment by bvandrc** [comment]
Also, per the a11y menu example, I would also think that `MenuItem` should have `role="menuitem"` and `Menu` should have `role="menu"` by default, but this is more of a recommendation, as I can currently manually pass these props to these items to have them take effect-- so this isn't necessarily a problem, only a reccomendation

**Comment by adidahiya** [comment]
Should we set `<li role="none">` always for MenuItem? From the example you linked, it seems like the answer is yes. And in addition, we should render `<a role="menuitem">`. But I'm just a little confused by this language, I'm not sure what "required list context" means:

> Necessary because the parent ul is serving as a menu so the li elements are not in their required list context.

**Comment by bvandrc** [comment]
> Should we set `<li role="none">` always for MenuItem? From the example you linked, it seems like the answer is yes. And in addition, we should render `<a role="menuitem">`. But I'm just a little confused by this language, I'm not sure what "required list context" means:
> 
> > Necessary because the parent ul is serving as a menu so the li elements are not in their required list context.

To answer your question, when you set `<ul role="menubar">` or  `<ul role="menu">`  , to a11y, the `ul` now serves as a different role than just the standard `ul`, so you can't just have regular `li` elements within it, because it sees those `li` items as not being within a `ul` or `ol`, since the `ul` has changed role. So the way to solve this is setting `<li role="none">` so that the `li` items, like the parent `ul`, now have a different role than being the standard `listitem`

**Comment by bvandrc** [comment]
> Should we set `<li role="none">` always for MenuItem? From the example you linked, it seems like the answer is yes. And in addition, we should render `<a role="menuitem">`.

I would say yes, that's what a11y is expecting. But if we do that we should maybe also have `Menu` have `role="menu"` by default with the option to set `role="menubar"` as well?

Then again, another solution of this would be to just add a `liRole` prop alongside `liClasses` in `MenuItem`. It depends on how much you want to include correct a11y roles in the blueprint elements by default-- that would be great, but allowing users to pass the roles themselves would satisfy my needs just the same.

**Comment by bvandrc** [comment]
created a PR for this
